.github/workflows/run-all-tests.yml

```
@@ -37,8 +37,8 @@ on:
     - cron: 0 3 * * 1 # run on 03:00 UTC every Monday
 
 concurrency:
-  # cancel older running jobs on the same branch
-  group: ${{ github.workflow }}-${{ github.ref }}
+  # cancel older running jobs on the same branch except release commit
+  group: ${{ (startsWith(github.event.head_commit.message, 'Prepare release') && github.ref) || format('{0}-{1}', github.workflow, github.ref) }}
```

```suggestion
  group: ${{ (startsWith(github.event.head_commit.message, 'Prepare release') && 'release-run' || format('{0}-{1}', github.workflow, github.ref) }}
```

If someone introduces a workflow that runs on every push to main in the future and uses the `github.ref` as the group, it might cancel a release run. Might be better to future proof this by adding a specific group to workflow runs.